servicetalk-loadbalancer-experimental-provider/src/main/java/io/servicetalk/loadbalancer/experimental/DefaultLoadBalancerObserver.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright © 2024 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.loadbalancer.experimental;
+
+import io.servicetalk.client.api.NoActiveHostException;
+import io.servicetalk.client.api.ServiceDiscovererEvent;
+import io.servicetalk.loadbalancer.LoadBalancerObserver;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import javax.annotation.Nullable;
+
+import static java.util.Objects.requireNonNull;
+
+final class DefaultLoadBalancerObserver implements LoadBalancerObserver {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultLoadBalancerObserver.class);
+
+    private final String clientName;
+
+    DefaultLoadBalancerObserver(final String clientName) {
+        this.clientName = requireNonNull(clientName, "clientName");
```

Comment:
Would be ideal if the `DefaultLoadBalancer` could propagate its `lbDescription`. The main motivation is being able to distinguish one instance of a client to "servicetalk.io" (or any "serviceFoo") from another instance to exact same address. Additionally, it will help to correlate observer events with `DefaultLoadBalancer` logs.